//> allow-more-errors

package Test {

    fn foo(){
        val f: Foo^#;
        enclosed {} {
            f = FooCreator.createFoo();
            abc(f.xs);      //> E : expected 'arr Bar', found 'arr^# Bar'
            bar(f.xs[0]);   //> E : expected 'Bar', found 'Bar^#'
        };
    }

    fn abc(arr Bar){}

    fn bar(Bar){}

}
